write.go

```diff
@@ -249,6 +249,15 @@ func (c *Conn) writeFrame(ctx context.Context, fin bool, flate bool, opcode opco
 	}
 	defer c.writeFrameMu.unlock()
 
+	if c.closeSent {
+		select {
+		case <-c.closed:
+			return 0, net.ErrClosed
+		default:
+		}
+		return 0, errors.New("close sent")
+	}
```

Comment:
Review: I took the liberty to make a few changes here since the code needed to move around.

Any message after the close frame could be considered bad behavior and wasted traffic, so I'm thinking we just don't allow it throughout.

Also, since we don't have a precedent in this library for sentinel errors, we'll just move it inline for now. I don't think sentinel errors are bad per-se, and maybe we should introduce some in the future (but let's not do that in this PR).

Lastly, an important detail here is that we're _not_ letting this context potentially close the connection before the handshake is complete by doing the check early.